src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs

```diff
@@ -1,408 +0,0 @@
-                        continue;
-                    }
-                }
-
-                if (!TryGetProjectDiagnosticsFromInMemoryStorage(serializerVersion, project, builder))
-                {
-                    // this can happen if SaveAsync is not yet called but active file merge happened. one of case is if user did build before the very first
-                    // analysis happened.
-                }
-
-                return builder.ToResult();
-            }
-
-            /// <summary>
-            /// Return all diagnostics for the given document stored in this state including non local diagnostics for this document
-            /// </summary>
-            public async Task<DiagnosticAnalysisResult> GetAnalysisDataAsync(TextDocument document, CancellationToken cancellationToken)
-            {
-                // make a copy of last result.
-                var lastResult = _lastResult;
-                Contract.ThrowIfFalse(lastResult.ProjectId == document.Project.Id);
-
-                var version = await GetDiagnosticVersionAsync(document.Project, cancellationToken).ConfigureAwait(false);
-                if (lastResult.IsDefault)
-                    return LoadInitialAnalysisData(document, version);
-
-                // if given document doesnt have any diagnostics, return empty.
-                if (IsEmpty(lastResult, document.Id))
-                {
-                    return DiagnosticAnalysisResult.CreateEmpty(lastResult.ProjectId, lastResult.Version);
-                }
-
-                // loading data can be canceled any time.
-                var serializerVersion = lastResult.Version;
-                var builder = new Builder(document.Project, lastResult.Version);
-
-                if (!TryGetDiagnosticsFromInMemoryStorage(serializerVersion, document, builder))
-                {
-                    Debug.Assert(lastResult.Version == VersionStamp.Default);
-
-                    // this can happen if we merged back active file diagnostics back to project state but
-                    // project state didn't have diagnostics for the file yet. (since project state was staled)
-                }
-
-                return builder.ToResult();
-            }
-
-            /// <summary>
-            /// Return all no location diagnostics for the given project stored in this state
-            /// </summary>
-            public async Task<DiagnosticAnalysisResult> GetProjectAnalysisDataAsync(Project project, CancellationToken cancellationToken)
-            {
-                // make a copy of last result.
-                var lastResult = _lastResult;
-                Contract.ThrowIfFalse(lastResult.ProjectId == project.Id);
-
-                var version = await GetDiagnosticVersionAsync(project, cancellationToken).ConfigureAwait(false);
-                if (lastResult.IsDefault)
-                    return LoadInitialProjectAnalysisData(project, version);
```

Comment:
lastResult.IsDefault would be true practically always. meaning we would virtually always go into LoadInitialProjectAnalysisData.